dask/dataframe/io/parquet.py

```diff
@@ -413,8 +420,16 @@ def _write_partition_pyarrow(df, open_with, filename, write_index,
 
     if metadata_path is not None:
         with open_with(metadata_path, 'wb') as fil:
-            kwargs.pop('compression', None)
-            parquet.write_metadata(t.schema, fil, **kwargs)
+            # Get only arguments specified in the function
+            try:
+                from inspect import getfullargspec as getargspec
+            except ImportError:
+                from inspect import getargspec
+            spec = getargspec(parquet.write_metadata)
+            safe_kwargs = {k: v for k, v in kwargs.items() if k in spec.args}
+            safe_kwargs['schema'] = t.schema
+            safe_kwargs['where'] = fil
+            parquet.write_metadata(**safe_kwargs)
```

Comment:
These args/kwargs to `write_metadata` change so infrequently (and arrow devs are active enough on dask) that I'd rather not rely on `getargspec`, but instead manually cull them down explicitly in the code.

Reply:
As commented in with `test_writing_parquet_with_unknown_kwargs`, in case we'd like to have a fixed number of arguments, it is better to make then into named arguments of function `_write-<engine>`. As for adding just explicit argument names list, it's fine but IMO it creates additional point of change if the underlying engine implementation changes.

Reply:
We use `kwargs` in `to_parquet`/`read_parquet` to specify arguments to the backend engines that differ between them. How the engines use these (via kwargs or explicit keywords) doesn't really matter, the user interface is the intention. In this case you could either:
- Make the `_write_*` take explicit keywords, and explicitly pass them to the underlying functions. This would be nicer, as we can error at call time instead of compute time for invalid parameters.
- Just filter the keyword arguments to `write_metadata`. This could be as simple as:

```python
metadata_kwargs = {'version', 'use_deprecated_int96_timestamps', 'coerce_timestamps'}
parquet.write_metadata(t.schema, fil, **{k: v for k, v in kwargs.items() if k in metadata_kwargs})
```

Reply:
Sure, I'll fix it. I'll give the first one a go as it'll be much more verbose for later use.